### PR TITLE
Fix SslConfig Merge When Using isolateVirtualHostsBySslConfig

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Fix a bug in the logic we have to consolidate SslConfiguration which are identical except for SNI.

# Context

During the investigation into the attached issue, @bdecoste identified this bug. We attempt to merge configuration that is identical (or overlapping) and in that merge, we were dropping Virtual Services. The impact is that a VirtualService would be accepted by our webhook, and during processing, it would not be persisted in the Proxy object, and therefore not appear in the Envoy config_dump.

The fix is rather trivial: if there is a list, append, if there is not, create a new one. We had that logic inverted.

## Testing
I would have expected unit tests to catch this, but there were none, so I added them. These tests fail if you undo the fix.

We have a number of e2e tests around this space, but they didn't cover the case of 2 overlapping sslConfigs. I updated the tests to cover this case. Similarly, if you ran these tests without the fix, they would fail.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/helpers/virtual_services.go [hunk 1]
```diff
@@ -208,6 +208,25 @@ func (b *virtualServiceBuilder) errorIfInvalid() error {
 	return nil
 }
 
+func (b *virtualServiceBuilder) Clone() *virtualServiceBuilder {
```
Comment (nfuden) [inline test/helpers/virtual_services.go:211]: Can we add a unit test to validate that this doesnt miss fields?

File: projects/gateway/pkg/translator/ssl_configuration_test.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+				vsEast = helpers.NewVirtualServiceBuilder().
+					WithName("vs-east").
+					WithNamespace(defaults.GlooSystem).
+					WithDomain("east.com").
+					WithSslConfig(&ssl.SslConfig{
+						OneWayTls: &wrappers.BoolValue{
+							Value: true,
+						},
+						SniDomains: []string{
+							"east.com",
```
Comment (nfuden) [inline projects/gateway/pkg/translator/ssl_configuration_test.go:74]: Lets also validate with multiple snidomains

File: test/helpers/virtual_services.go (hunk 1 above)
Comment (sam-heilbron) [reply test/helpers/virtual_services.go:211]: Yeah good call, I had added it for the other two but had neglected to do it for this builder, ill update

Comment (sam-heilbron) [reply test/helpers/virtual_services.go]: https://github.com/solo-io/gloo/pull/7999/commits/e97444facb22b20e190af406194c9503a9b4aee6

File: projects/gateway/pkg/translator/ssl_configuration_test.go (hunk 2 above)
Comment (sam-heilbron) [reply projects/gateway/pkg/translator/ssl_configuration_test.go:76]: https://github.com/solo-io/gloo/commit/e7e0d824de16e9b7959e6dfd01151fc049024ec7 I updated the existing test to include multiple values, some of which were overlapping

File: test/helpers/helpers_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	"testing"
+
+	"github.com/solo-io/go-utils/testutils"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+func TestHelpers(t *testing.T) {
+	testutils.RegisterCommonFailHandlers()
+	RunSpecs(t, "Helpres Suite")
```
Comment (sam-heilbron) [inline test/helpers/helpers_suite_test.go:13]: typo, fix!

**Review by nfuden — APPROVED** [review]

**Review by gunnar-solo — APPROVED** [review]
make it so.

**Pull request merged**